src/game/client/draw/w3dmodeldraw.cpp

```diff
@@ -1154,3 +1179,1914 @@ void W3DModelDrawModuleData::Parse_Condition_State(INI *ini, void *instance, voi
+            subobj.visible = info.visible;
+            xfer->xferBool(&subobj.visible);
+        }
+    } else {
+        m_subObjects.clear();
+        for (int j = 0; j < size; j++) {
+            xfer->xferAsciiString(&subobj.name);
+            xfer->xferBool(&subobj.visible);
+            m_subObjects.push_back(subobj);
+        }
+    }
+
+    if (ver >= 2) {
+        if (xfer->Get_Mode() == XFER_SAVE) {
+            if (m_renderObject != nullptr && m_renderObject->Class_ID() == RenderObjClass::CLASSID_HLOD
+                && m_curState != nullptr && m_curState->m_transition) {
+                HLodClass *hlod = static_cast<HLodClass *>(m_renderObject);
+                float frame;
+                int frames;
+                int mode;
+                float multiplier;
+                HAnimClass *anim = hlod->Peek_Animation_And_Info(frame, frames, mode, multiplier);
+                bool b = anim != nullptr;
+                xfer->xferBool(&b);
+
+                if (anim) {
+                    xfer->xferInt(&mode);
+                    float f = frame / (anim->Get_Num_Frames() - 1);
+                    xfer->xferReal(&f);
+                }
+            } else {
+                bool b = false;
+                xfer->xferBool(&b);
+            }
+        } else {
+            bool b;
+            xfer->xferBool(&b);
+
+            if (b) {
+                int mode;
+                float newframe;
+                xfer->xferInt(&mode);
+                xfer->xferReal(&newframe);
+
+                if (m_renderObject != nullptr) {
+                    if (m_renderObject->Class_ID() == RenderObjClass::CLASSID_HLOD) {
+                        HLodClass *hlod = static_cast<HLodClass *>(m_renderObject);
+                        HAnimClass *anim = hlod->Peek_Animation();
+
+                        if (anim) {
+                            float frame;
+                            int frames;
+                            int mode;
+                            float multiplier;
+                            hlod->Peek_Animation_And_Info(frame, frames, mode, multiplier);
+                            hlod->Set_Animation(anim, (anim->Get_Num_Frames() - 1) / newframe, mode);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    if (xfer->Get_Mode() == XFER_LOAD && !m_subObjects.empty()) {
+        Update_Sub_Objects();
+    }
+}
+
+void W3DModelDraw::Load_Post_Process()
+{
+    DrawModule::Load_Post_Process();
+}
+
+NameKeyType W3DModelDraw::Get_Module_Name_Key() const
+{
+    static NameKeyType nk = g_theNameKeyGenerator->Name_To_Key("W3DModelDraw");
+    return nk;
+}
+
+void W3DModelDraw::Show_Sub_Object(Utf8String const &sub_obj_name, bool visible)
+{
+    if (sub_obj_name.Is_Not_Empty()) {
+        bool found = false;
+
+        for (auto &info : m_subObjects) {
+            if (!strcasecmp(sub_obj_name, info.name)) {
+                info.visible = visible == 0;

```

Comment:
Could it be that `info.visible` should be `info.hidden` instead? Because `visible == 0` is false. Also you can write `!visible` here for boolean.